### PR TITLE
JDK15+ bindNative() attempts the systemClassLoader nativeLibrary first

### PR DESCRIPTION
**Pull request by JasonFengJ9**
`JDK15+` attempts to bind the nativeMethod by looking through the native libraries in `systemClassLoader` first;
Added a helper method `bindNativeClassLoader()`.

closes https://github.com/eclipse/openj9/issues/11512

Signed-off-by: Jason Feng <fengj@ca.ibm.com>

### CODE REVIEW

**Comment by pshipton** [comment]
Not sure this should close #11512 as per https://github.com/eclipse/openj9/issues/11512#issuecomment-748541108

Is it confirmed the reference has the same behavior?

**Comment by JasonFengJ9** [comment]
> Is it confirmed the reference has the same behavior?

As per https://github.com/eclipse/openj9/issues/11512#issuecomment-748545476, the issue occurred when the module `openj9.sharedclasses` class `com.ibm.oti.shared.Shared` was loaded by `jdk.internal.loader.ClassLoaders$AppClassLoader` via `ServiceLoader`. There is no straightforward comparison with RI.

**Comment by pshipton** [comment]
> There is no straightforward comparison with RI.

We need to create a testcase to confirm the behavior.

File: runtime/vm/bindnatv.cpp [hunk 1]
```diff
@@ -958,6 +959,41 @@ resolveNativeAddress(J9VMThread *currentThread, J9Method *nativeMethod, UDATA ru
...
+ * @param[in] bindJNINative non-zero if JNI natives are allowed, zero for INL-only
+ *
+ * @return J9_NATIVE_METHOD_BIND_SUCCESS on success,
+ *         J9_NATIVE_METHOD_BIND_FAIL on failure.
+ */
+static UDATA
+bindNativeClassLoader(J9VMThread *currentThread, J9ClassLoader *classLoader, J9Method *nativeMethod, char *longJNI, char *shortJNI, U_8 argCount, UDATA bindJNINative)
+{
+	/* Search each shared library in the class loader for a matching native */
+	UDATA rc = J9_NATIVE_METHOD_BIND_FAIL;
```
Comment (pshipton) [inline runtime/vm/bindnatv.cpp:980]: This can be declared in the inner scope.

File: runtime/vm/bindnatv.cpp [hunk 2]
```diff
@@ -958,6 +959,41 @@ resolveNativeAddress(J9VMThread *currentThread, J9Method *nativeMethod, UDATA ru
...
+	J9NativeLibrary *nativeLibrary = classLoader->librariesHead;
+	while (NULL != nativeLibrary) {
+		rc = lookupNativeAddress(currentThread, nativeMethod, nativeLibrary, longJNI, shortJNI, argCount, bindJNINative);
+		if (J9_NATIVE_METHOD_IS_BOUND(nativeMethod)) {
```
Comment (pshipton) [inline runtime/vm/bindnatv.cpp:984]: Is `rc` guaranteed to be J9_NATIVE_METHOD_BIND_SUCCESS? This is a change from the original code which has a hard coded return of J9_NATIVE_METHOD_BIND_SUCCESS.

Comment (JasonFengJ9) [reply runtime/vm/bindnatv.cpp:984]: https://github.com/eclipse/openj9/blob/f0a4235ef0e2e6e1a8d476c4cf8dbf1ea1bc1cdf/runtime/vm/bindnatv.cpp#L1170-L1179
`rc` returned should be `J9_NATIVE_METHOD_BIND_SUCCESS` when the `nativeMethod` is bound successfully.

File: runtime/vm/bindnatv.cpp (hunk 1 above)
Comment (JasonFengJ9) [reply runtime/vm/bindnatv.cpp:980]: This helper method returns `rc` value hence this is right scope.

**Review by pshipton — CHANGES_REQUESTED** [review]
I don't think this is correct, and the problem should be resolved another way. See https://github.com/eclipse/openj9/issues/11512#issuecomment-749055413

**Comment by JasonFengJ9** [comment]
https://github.com/eclipse/openj9/issues/11512 has been closed via https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/260.
Though `JDK15+` `bindNative()` probably should attempt the systemClassLoader nativeLibrary first like this PR, I won't proceed without a valid usecase.
closing.

**Pull request closed without merging**